### PR TITLE
Update go and workflows

### PR DESCRIPTION
**Pull request by k1LoW**
Hi @linyows.

Updated Go version and workflows for GitHub Actions.

Also, since golint has been deprecated, golanci-lint has been installed instead.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,15 +13,15 @@ on:
...
         os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-go@v2
+    - uses: actions/checkout@v4
+    - uses: actions/setup-go@v4
       with:
-        go-version: ${{ matrix.go-version }}
+        go-version-file: go.mod
```
Comment (k1LoW) [inline .github/workflows/build.yml:24]: Changed to test only the version of Go specified in go.mod

**Comment by linyows** [comment]
Many thanks!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -13,15 +13,15 @@ on:
 jobs:
   build:
     name: Build
-    runs-on: ubuntu-latest
     strategy:
       matrix:
-        go-version: [1.13.x, 1.14.x, 1.15.x, 1.16.x]
         os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
```
Comment (k1LoW) [inline .github/workflows/build.yml:19]: Now using matrix.os

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+linters:
+  fast: false
+  enable:
```
Comment (k1LoW) [inline .golangci.yml:3]: Enabled linters are my preference, so feel free to change it!

**Pull request merged**